test/com/twu/biblioteca/BibliotecaAppTest.java

```diff
@@ -0,0 +1,62 @@
+package com.twu.biblioteca;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+public class BibliotecaAppTest {
+
+    @Test //[#1.2 & #1.3]
+    public void displayAvailableBooksCalled() {
+        DummyMainMenuInterface mainMenuInterface = new DummyMainMenuInterface();
+
+        mainMenuInterface.displayAvailableBooks();
+
+        assertTrue(mainMenuInterface.displayAvailableBooksHasBeenCalled);
+    }
```

Comment:
I know this is a work-in-progress, so feel free to ignore this if it's going to change in subsequent commits, but in the test for `BibliotecaApp` you should naturally be creating an instance of `BibliotecaApp` and calling methods on that instance.  In this test, you are creating an instance of `DummyMainMenuInterface` and calling methods on that instance instead.  

The assertion is correct, but the test should be checking how that method has been called _indirectly_ via `BibliotecaApp`, rather than testing it directly.

Let me know if this does/doesn't make sense and I can jump on a quick call with you.

Reply:
I'll have to talk to you tomorrow. I tried to create an instance of BibliotecaApp in the test but intelliJ didn't like it due to the static methods used in BibliotecaApp.

Reply:
I tried to implement the following but not sure if I'm making the correct connections between the classes:

`@Test`
`public void displayAvailableBooksCalled() {`
       ` BibliotecaApp bibliotecaApp = new BibliotecaApp();`
        `bibliotecaApp.manageOptions(DummyMainMenu dummyMainMenu);`
       ` dummyMainMenu.displayAvailableBooks();`
        `assertTrue(dummyMainMenu.displayAvailableBooksHasBeenCalled);`
   ` }`

Reply:
That's _almost_ there!  You do not need to call `displayAvailableBooks` from your test as `BibliotecaApp` will be doing it for you when you ask it to `manageOptions`.  If you compare this to the sample one I sent through:

```
        Parent parent = new Parent();
        TestChild child = new TestChild();
        parent.doesStuff(child);

        assertTrue(child.hasBeenCalled);
```
* `parent` is the equivalent of your `bibliotecaApp`
* `child` is the equivalent of your `dummyMainMenu`
* `parent.doesStuff` is the equivalent of `bibliotecaApp.manageOptions`.

Reply:
I tried the below but the test runs and never completes, it just keeps running:

      
```
        BibliotecaApp bibliotecaApp = new BibliotecaApp();
        DummyMainMenuInterface dummyMainMenu = new DummyMainMenuInterface();
        bibliotecaApp.manageOptions(dummyMainMenu);

        assertTrue(dummyMainMenu.displayAvailableBooksHasBeenCalled);
```

Reply:
That is exactly how I would expect the test to look, so the fact the test never concludes suggests the loop is continuing to run... there are a couple of ways to address this:

1. Extract the switch statement into a method that only takes the user input and then call that method in your test (so you isolate the test from the loop), or
1. Have your test provide the keypresses to simulate the pressing of keys needed to exit the loop.

Looking through this stuff, I think Option 1 is an easier option to implement.